scripts-dev/complement.sh

```diff
@@ -65,4 +65,4 @@ if [[ -n "$1" ]]; then
 fi
 
 # Run the tests!
```

Comment:
Running on this branch, https://github.com/matrix-org/complement/pull/214

The tests are looking good.

 - The existing room version tests are failing as expected because we dropped support for them
 - Unrecognized event test is failing but can easily fix that up in another PR with whatever error we want to assert now for that.

```
$ COMPLEMENT_ALWAYS_PRINT_SERVER_LOGS=1 COMPLEMENT_DIR=../complement ./scripts-dev/complement.sh TestImportHistoricalMessages

--- FAIL: TestImportHistoricalMessages (64.59s)
    --- FAIL: TestImportHistoricalMessages/parallel (0.67s)
        --- SKIP: TestImportHistoricalMessages/parallel/TODO:_Trying_to_send_insertion_event_with_same_`next_batch_id`_will_reject (0.00s)
        --- SKIP: TestImportHistoricalMessages/parallel/TODO:_What_happens_when_you_point_multiple_batches_at_the_same_insertion_event? (0.00s)
        --- FAIL: TestImportHistoricalMessages/parallel/Existing_room_versions (0.00s)
            --- FAIL: TestImportHistoricalMessages/parallel/Existing_room_versions/Not_allowed_to_redact_MSC2716_insertion,_batch,_marker_events (0.57s)
            --- FAIL: TestImportHistoricalMessages/parallel/Existing_room_versions/Room_creator_can_send_MSC2716_events (0.67s)
        --- PASS: TestImportHistoricalMessages/parallel/Normal_users_aren't_allowed_to_batch_send_historical_messages (2.41s)
        --- PASS: TestImportHistoricalMessages/parallel/Unrecognised_batch_id_will_throw_an_error (2.43s)
        --- PASS: TestImportHistoricalMessages/parallel/Batch_send_endpoint_only_returns_state_events_that_we_passed_in_via_state_events_at_start (3.46s)
        --- PASS: TestImportHistoricalMessages/parallel/Should_be_able_to_send_a_batch_without_any_state_events_at_start_-_user_already_joined_in_the_current_room_state (3.86s)
        --- PASS: TestImportHistoricalMessages/parallel/Should_be_able_to_batch_send_historical_messages_into_private_room (4.69s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_from_multiple_users_in_the_same_batch (5.04s)
        --- FAIL: TestImportHistoricalMessages/parallel/Unrecognised_prev_event_ID_will_throw_an_error (2.08s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_resolve_in_the_correct_order (6.66s)
        --- PASS: TestImportHistoricalMessages/parallel/Duplicate_next_batch_id_on_insertion_event_will_be_rejected (2.81s)
        --- PASS: TestImportHistoricalMessages/parallel/should_resolve_member_state_events_for_historical_events (7.38s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_from_/batch_send_do_not_come_down_in_an_incremental_sync (6.09s)
        --- PASS: TestImportHistoricalMessages/parallel/When_messages_have_already_been_scrolled_back_through,_new_historical_messages_are_visible_in_next_scroll_back_on_federated_server (7.40s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_messages_are_visible_when_joining_on_federated_server_-_auto-generated_base_insertion_event (9.84s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_messages_are_visible_when_joining_on_federated_server_-_pre-made_insertion_event (9.90s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_messages_are_visible_when_already_joined_on_federated_server (10.67s)
        --- PASS: TestImportHistoricalMessages/parallel/Backfill_still_works_after_many_batches_are_imported (55.51s)
FAIL
FAIL	github.com/matrix-org/complement/tests	65.066s
FAIL
```

Reply:
To avoid confusion, we support current room versions again. Here are the Complement test results nowadays:

```
--- FAIL: TestImportHistoricalMessages (67.41s)
    --- FAIL: TestImportHistoricalMessages/parallel (54.62s)
        --- SKIP: TestImportHistoricalMessages/parallel/TODO:_Trying_to_send_insertion_event_with_same_`next_batch_id`_will_reject (0.00s)
        --- SKIP: TestImportHistoricalMessages/parallel/TODO:_What_happens_when_you_point_multiple_batches_at_the_same_insertion_event? (0.00s)
        --- PASS: TestImportHistoricalMessages/parallel/Federation (0.00s)
            --- PASS: TestImportHistoricalMessages/parallel/Federation/Historical_messages_are_visible_when_joining_on_federated_server_-_auto-generated_base_insertion_event (5.63s)
            --- PASS: TestImportHistoricalMessages/parallel/Federation/Historical_messages_are_visible_when_joining_on_federated_server_-_pre-made_insertion_event (6.76s)
            --- PASS: TestImportHistoricalMessages/parallel/Federation/When_messages_have_already_been_scrolled_back_through,_new_historical_messages_are_visible_in_next_scroll_back_on_federated_server (7.47s)
            --- PASS: TestImportHistoricalMessages/parallel/Federation/Historical_messages_are_visible_when_already_joined_on_federated_server (7.52s)
            --- PASS: TestImportHistoricalMessages/parallel/Federation/Backfill_still_works_after_many_batches_are_imported (53.55s)
        --- PASS: TestImportHistoricalMessages/parallel/Existing_room_versions (0.00s)
            --- PASS: TestImportHistoricalMessages/parallel/Existing_room_versions/Not_allowed_to_redact_MSC2716_insertion,_batch,_marker_events (0.98s)
            --- PASS: TestImportHistoricalMessages/parallel/Existing_room_versions/Room_creator_can_send_MSC2716_events (1.06s)
        --- FAIL: TestImportHistoricalMessages/parallel/Unrecognised_prev_event_ID_will_throw_an_error (1.49s)
        --- PASS: TestImportHistoricalMessages/parallel/Duplicate_next_batch_id_on_insertion_event_will_be_rejected (2.12s)
        --- PASS: TestImportHistoricalMessages/parallel/Normal_users_aren't_allowed_to_batch_send_historical_messages (2.27s)
        --- PASS: TestImportHistoricalMessages/parallel/Unrecognised_batch_id_will_throw_an_error (2.28s)
        --- PASS: TestImportHistoricalMessages/parallel/Batch_send_endpoint_only_returns_state_events_that_we_passed_in_via_state_events_at_start (2.82s)
        --- PASS: TestImportHistoricalMessages/parallel/Should_be_able_to_send_a_batch_without_any_state_events_at_start_-_user_already_joined_in_the_current_room_state (3.23s)
        --- PASS: TestImportHistoricalMessages/parallel/Should_be_able_to_batch_send_historical_messages_into_private_room (3.68s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_from_multiple_users_in_the_same_batch (3.73s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_from_batch_send_do_not_come_down_in_an_incremental_sync (4.18s)
        --- PASS: TestImportHistoricalMessages/parallel/Historical_events_resolve_in_the_correct_order (4.26s)
        --- PASS: TestImportHistoricalMessages/parallel/should_resolve_member_state_events_for_historical_events (4.28s)
FAIL
FAIL	github.com/matrix-org/complement/tests	67.916s
```